llama/runner/cache.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"log/slog"
+	"time"
+
+	"github.com/ollama/ollama/llama"
+)
+
+type TokenCache struct {
+	// context window size (per slot)
```

Comment:
This comment can be dismissed/resolved if not, but do we need to any kind of locking here or in TokenCacheSlot?

Reply:
We need to hold a lock when performing operations on TokenCacheSlot or llama.Decode, since it also manipulates the KV cache. We are doing this currently with Server.mu. I added a comment to document this.

This is adding more code under a big lock, which is not ideal though the cache lookup is fast. I still hope that we can break it up but this is safe and straightforward for now.

Reply:
Great! Thanks